flume-ng-sinks/flume-hdfs-sink/src/test/java/org/apache/flume/sink/hdfs/TestHDFSEventSink.java

```diff
@@ -1323,6 +1323,64 @@ public void testCloseOnIdle() throws IOException, EventDeliveryException, Interr
     fs.close();
   }
 
+  @Test
+  public void testBlockCompressSequenceFileWriterSync() throws IOException, EventDeliveryException, InterruptedException {
+    String hdfsPath = testPath + "/sequenceFileWriterSync";
+    String body = "test event-" + System.nanoTime();
+
+    Configuration conf = new Configuration();
+    FileSystem fs = FileSystem.get(conf);
+
+    // Since we are reading a partial file we don't want to use checksums
+    fs.setVerifyChecksum(false);
+    fs.setWriteChecksum(false);
+
+    Path dirPath = new Path(hdfsPath);
+    fs.delete(dirPath, true);
+    fs.mkdirs(dirPath);
+
+    Context context = new Context();
+    context.put("hdfs.path", hdfsPath);
+    // Ensure the file isn't closed and rolled, we are only going to send one event
+    context.put("hdfs.rollCount", "10");
+    context.put("hdfs.rollSize", "0");
+    context.put("hdfs.rollInterval", "0");
+    context.put("hdfs.batchSize", "1");
+    context.put("hdfs.fileType", "SequenceFile");
+    context.put("hdfs.codeC", "BZip2Codec"); // Compression codec that doesn't require native hadoop libraries
+    context.put("hdfs.writeFormat", "Writable");
+    Configurables.configure(sink, context);
+
+    Channel channel = new MemoryChannel();
+    Configurables.configure(channel, context);
+
+    sink.setChannel(channel);
+    sink.start();
+
+    Transaction txn = channel.getTransaction();
+    txn.begin();
+    Event event = new SimpleEvent();
+    event.setBody(body.getBytes());
+    channel.put(event);
```

Comment:
The test is pretty good but please make the following modifications:

1. Add multiple events to the channel.
2. Call process() multiple times.
3. Ensure that the file is readable by a reader after multiple batches have been flushed.
4. Test all of the following codecs: bzip2, gzip, snappy

You probably would want to extract a helper method to avoid copy / paste for testing all the codecs